src/shape.rs

```diff
@@ -1194,9 +1204,15 @@ impl ShapeLine {
                             }
                             word_range_width += word_width;
                             continue;
-                        } else if wrap == Wrap::Glyph
+                        }
+
+                        if wrap == Wrap::Glyph
```

Comment:
Is this piece intended? Is it always the case that this case (glyph-wrapping) is now NOT mutually exclusive of the case above i.e. we're switching from `else if` to `if` here?

Will we be following similar logic when implementing this for incongruent directions later (noticed a similar block there)?

Reply:
Yes, the `if` case ends in `continue`, so the two are mutually exclusive. I personally think it's more readable to have this as a separate `if` statement in this case, but I'm open to changing it!

I haven't looked at the incongruent case yet, but yes, I think it'd be similar.

Reply:
ah, missed the `continue` - agreed, aligned!